packages/@react-spectrum/dialog/src/Dialog.tsx

```diff
@@ -78,14 +78,16 @@ let sizeMap = {
 
 function BaseDialog({children, slots, size, role, ...otherProps}: SpectrumBaseDialogProps) {
   let ref = useRef();
+  let titleId = useSlotId();
+  titleId = otherProps['aria-label'] ? undefined : titleId;
```

Comment:
The logic here and below (setting aria-labelledby) should go in `useDialog`.

Reply:
good catch, thanks